luarules/gadgets/unit_water_depth_damage.lua

```diff
@@ -0,0 +1,173 @@
+function gadget:GetInfo()
+	return {
+		name = "Water Crush and Collision Damage",
+		desc = "Creates and handles water collision events, and kills units stuck underwater",
+		author = "SethDGamre",
+		date = "2024.9.22",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+--use customParams.water_fall_damage_multiplier = 1.0 to change the amount of fall damage taken by specific units.
+
+--required velocity in a frame for a unit to take collision damage from falling into water.
+local velocityThreshold = 99 / Game.gameSpeed
+
+--any maxWaterDepth movedef equal to or above this number will not take drowning damage.
+-- performance optimisation to avoid checking amphs and hovers
+local isDrownableMaxWaterDepth = 5000
+
+--a percentage of health taken as damage per second when stuck below maxWaterDepth.
+local drowningDamage = 0.05
```

Comment:
Not terribly familiar with the game, so genuine curiosity question here, is percentage damage appropriate, and is this the appropriate threshold?

Reply:
I made it up. Nobody will care because this damage method will occur in edge cases where a unit gets stuck where it doesn't belong underwater.

Percentage is appropriate because it'll kill all units at same speed when stuck.

This is configurable easily in the future if someone wants to actually make drowning damage a more significant feature in the game.